gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapAdapter.java

```diff
@@ -132,6 +161,56 @@ public void moveCamera(Point point, float zoomLevel) {
     map.moveCamera(CameraUpdateFactory.newLatLngZoom(point.toLatLng(), zoomLevel));
   }
 
+  public void renderJsonLayer() {
+    File file = new File(context.getFilesDir(), GEO_JSON_FILE);
+    if (file.exists()) {
+      try {
+        InputStream is = new FileInputStream(file);
+        BufferedReader buf = new BufferedReader(new InputStreamReader(is));
+        String line = buf.readLine();
+        StringBuilder sb = new StringBuilder();
+        while (line != null) {
+          sb.append(line).append("\n");
+          line = buf.readLine();
+        }
+        try {
+          JSONObject geoJson = new JSONObject(sb.toString());
+          GeoJsonLayer layer = new GeoJsonLayer(map, geoJson);
+          layer.setOnFeatureClickListener(this::onJsonFeatureClick);
+          layer.addLayerToMap();
+          Log.d(TAG, "JSON successfully loaded.");
+        } catch (JSONException e) {
+          Log.d(TAG, "Unable to read JSON.", e);
+          e.printStackTrace();
+        }
+      } catch (IOException e) {
+        Log.d(TAG, "geojson file not found.", e);
+      }
+    }
+  }
+
+  private void onJsonFeatureClick(com.google.maps.android.data.Feature feature) {
```

Comment:
Knowledge of the Ground-specific VOs don't really belong here since this class is lower level; it just exposes functionality of Maps API or whatever implementation we use in a generic way, so it covers only mapping concepts like Points, Lines, Polygons, etc. GeoJSON may fix in this context, but we'd ideally find a way to generalize it to not scope-creep the interface needed to be implemented for new map providers.